### PR TITLE
fix: SPM macOS/tvOS disable event explorer

### LINKED ISSUE #298: Error building 7.1.0 on Xcode 12.2 beta 3 for tvOS
I've included Amplitude-iOS using SPM. I just updated from 5.2.0 (which built just fine) to 7.1.0, which no longer builds.

I get the following build errors:

SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/AMPInfoViewController.m:69:46: 'UIPasteboard' is unavailable: not available on tvOS
/Applications/Xcode-beta.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator14.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIPasteboard.h:32:96: 'UIPasteboard' has been explicitly marked unavailable here
SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/AMPInfoViewController.m:70:16: 'string' is unavailable: not available on tvOS
/Applications/Xcode-beta.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator14.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIPasteboard.h:90:46: 'string' has been explicitly marked unavailable here
SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/AMPInfoViewController.m:76:46: 'UIPasteboard' is unavailable: not available on tvOS
/Applications/Xcode-beta.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator14.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIPasteboard.h:32:96: 'UIPasteboard' has been explicitly marked unavailable here
SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/AMPInfoViewController.m:77:16: 'string' is unavailable: not available on tvOS
/Applications/Xcode-beta.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator14.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIPasteboard.h:90:46: 'string' has been explicitly marked unavailable here
SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/AMPEventExplorer.m:88:13: Initializing 'CGFloat' (aka 'double') with an expression of incompatible type 'id'
SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/AMPEventExplorer.m:96:10: Assigning to 'CGFloat' (aka 'double') from incompatible type 'void'
ARC Semantic Issue Group
SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/AMPEventExplorer.m:53:44: No known class method for selector 'barBottomOffset'
SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/AMPEventExplorer.m:88:41: No known class method for selector 'statusBarHeight'
SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/AMPEventExplorer.m:89:40: No known class method for selector 'barBottomOffset'

### PR DESCRIPTION
**Pull request by jooohhn**
### Summary

- Fixes #298 
  - Package.swift conditionally uses EventExplorer files only on iOS platform
- Reorganize project folder structure to reflect XCode structure and SPM requirements 
- Tested on
  - iOS + SPM project
  - tvOS + SPM project
  - macOS + SPM project

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-iOS/blob/master/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:   NO

### CODE REVIEW

**Review by haoliu-amp — COMMENTED** [review]
Do we need `.build/manifest.db` checked in?

File: Amplitude.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -815,18 +819,18 @@
...
-				12EAC0322547D42B00382C8A /* cancel.png in Resources */,
-				12EAC02F2547D42B00382C8A /* logo-button@2x.png in Resources */,
-				12EAC02C2547D42B00382C8A /* cancel@3x.png in Resources */,
-				12EAC0312547D42B00382C8A /* logo-banner.png in Resources */,
-				12EAC03F2547D44300382C8A /* ComodoRsaDomainValidationCA.der in Resources */,
-				12EAC0342547D42B00382C8A /* logo-button@3x.png in Resources */,
-				12EAC02E2547D42B00382C8A /* logo-button.png in Resources */,
-				12EAC03E2547D44300382C8A /* AMPBubbleView.xib in Resources */,
-				12EAC0402547D44300382C8A /* AMPInfoViewController.xib in Resources */,
+				29E7B667255E487100AB9A5B /* logo-button@3x.png in Resources */,
```
Comment (haoliu-amp) [inline Amplitude.xcodeproj/project.pbxproj:822]: What's the motivation to move these files? Make it look better?

File: Package.swift [hunk 2]
```diff
@@ -14,24 +14,34 @@ let package = Package(
...
+                .process("Resources/Images/logo-banner@2x.png"),
+                .process("Resources/Images/logo-banner@3x.png"),
+                .process("Resources/Images/logo-button.png"),
+                .process("Resources/Images/logo-button@2x.png"),
+                .process("Resources/Images/logo-button@3x.png")
             ],
-            publicHeadersPath: ".")
+            publicHeadersPath: ".",
+            cSettings: [
+                .headerSearchPath("../Amplitude")
```
Comment (haoliu-amp) [inline Package.swift:35]: What's this for?

File: Package.swift [hunk 3]
```diff
@@ -14,24 +14,34 @@ let package = Package(
...
-                .process("Resources/cancel@2x.png"),
-                .process("Resources/cancel@3x.png"),
-                .process("Resources/ComodoRsaDomainValidationCA.der"),
-                .process("Resources/logo-banner.png"),
-                .process("Resources/logo-banner@2x.png"),
-                .process("Resources/logo-banner@3x.png"),
-                .process("Resources/logo-button.png"),
-                .process("Resources/logo-button@2x.png"),
-                .process("Resources/logo-button@3x.png")
+                .process("Resources/Images/cancel.png"),
```
Comment (haoliu-amp) [inline Package.swift:23]: When you move these files, make sure the `Target Membership` for them is not changed in xcode. They (EE related) should be only for iOS.

**Review by haoliu-amp — COMMENTED** [review]
Do we need `.build/manifest.db` checked in?

**Comment by jooohhn** [comment]
> Do we need `.build/manifest.db` checked in?

I think it's an xcode related thing, should we add it to gitignore?

File: Package.swift (hunk 2 above)
Comment (jooohhn) [reply Package.swift:35]: [`EventExplorer` uses files outside its group](https://github.com/Amplitude/Amplitude-iOS/blob/macOS-tvOS-disable-event-explorer/Sources/EventExplorer/AMPInfoViewController.m#L26). If we don't configure `headerSearchPath`, then `EventExplorer` won't be able to find them


Build failure example with no `headerSearchPath`:

image

File: Package.swift (hunk 3 above)
Comment (jooohhn) [reply Package.swift:23]: this was not changed so we should be good

**Review by haoliu-amp — APPROVED** [review]

**Pull request merged**